proposals/0052-svg-components.md

```diff
@@ -0,0 +1,247 @@
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<Logo width={48} height={48} />
+```
+
+## Inline Implementation
+
+Each time a `.svg` file is imported and rendered in Astro, Astro will directly inject the optimized `.svg` file contents into the location with the root `<svg>` element properties overridden by the props passed to the SVG Component.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<Logo width={48} height={48} />
+```
+
+This would generate the following output:
+
+```astro
+<svg width="48" height="48">
+    <!-- SVG Content -->
+</svg> 
+```
+
+## Sprite Implementation
+
+When a `.svg` file is first imported and rendered in Astro, the system will convert the file into a `<symbol>` element. This will be inserted into the `<svg>` element. This approach ensures that all subsequent renders of that `.svg` can use a `<use>` element, referencing the ID of the initial `<symbol>`.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<!-- First Usage -->
+<Logo width={48} height={48} />
+
+<!-- Second Usage -->
+<Logo width={48} height={48} />
+
+```
+
+This would generate the following output:
+
+```astro
+<svg width="48" height="48">
+    <symbol id="a:0">
+        <!-- SVG Content -->
+    </symbol>
+    <use href="#a:0" />
+</svg> 
+<svg width="48" height="48">
+    <use href="#a:0" />
+</svg> 
+```
+
+This pattern significantly reduces the amount of duplicated inline SVG markup and speeds up the browser rendering by reducing the number of DOM nodes.
+
+## Prop Inheritance and Overriding
+
+When a `.svg` is treated as a component, users can pass attributes such as `width`, `height`, `fill`, `stroke`, and other common SVG attributes. These attributes will automatically be applied to the root `<svg>` element unless they conflict with existing properties in the original `.svg` file, in which case the passed props will override the existing ones.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg'
+---
+
+<!-- Pass width and height to override default size -->
+<Logo width={100} height={100} fill="blue" />
+```
+
+### Sizing
+
+To simplify the process of setting the dimensions of SVG components, this proposal introduces a `size` prop that developers can use to uniformly scale the width and height of an SVG. This prop provides a convenient shorthand for situations where both width and height need to be set to the same value, offering a more concise API.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<!-- Using the size prop to set both width and height -->
+<Logo size={48} />
+```
+
+This would generate the following output:
+
+```astro
+<svg width="48" height="48">
+    <!-- SVG Content -->
+</svg> 
```

Comment:
I believe we should go in more detail, and explain how the component will behave with `width` and `height`. For example, given these two cases where we have the same props in a different order, what is the end result? 

```jsx
<>
	<Logo size={50} width={20} />
	<Logo width={20} size={50} />
</>
```

I believe it's also fine to consider an error if the two props are incompatible.

Reply:
Definitely agree this deserves more detail around a somewhat tricky prop. I left this as an outstanding question since I wasn't sure the best way to handle the overlap of height/width props & attributes, the size prop and intrinsic size data coming from the image metadata. https://github.com/withastro/roadmap/pull/1035/files#diff-ab3122e95078e177eab4f364380c44e582e73c496b98809d7d5e966f9d09e1f1R247

Reply:
Whatever logic we decide is best, it is probably worth adding a warning around it if we decide not to apply.